synapse/rest/admin/users.py

```diff
@@ -461,7 +461,14 @@ async def on_POST(self, request):
 
 
 class WhoisRestServlet(RestServlet):
-    PATTERNS = historical_admin_path_patterns("/whois/(?P<user_id>[^/]*)")
+    path_regex = "/whois/(?P<user_id>[^/]*)"
+    PATTERNS = (
+        admin_patterns(path_regex)
+        +
+        # URL for spec reason
+        # https://matrix.org/docs/spec/client_server/r0.6.1#get-matrix-client-r0-admin-whois-userid
+        [re.compile("/_matrix/client/r0/admin" + path_regex)]
```

Comment:
I suspect that this should use `client_patterns("/admin$", v1=True)` to ensure this ends up at all the old paths:

* `/_matrix/client/api/v1/admin`
* `/_matrix/client/unstable/admin`
* `/_matrix/client/r0/admin`

This can be imported: `from synapse.rest.client.v2_alpha._base import client_patterns`